tools/autograd/derivatives.yaml

```diff
@@ -1019,13 +1019,13 @@
 - name: rsqrt(Tensor self) -> Tensor
   self: -0.5 * grad * result.pow(3).conj()
 
-- name: scatter_.src(Tensor(a!) self, int dim, Tensor index, Tensor src) -> Tensor(a!)
-  self: grad.clone().scatter_(dim, index, 0)
+- name: scatter.src(Tensor self, int dim, Tensor index, Tensor src, *, str? reduce=None) -> Tensor
+  self: scatter_self_backward(grad.clone(), dim, index, src, reduce)
   index: non_differentiable
-  src: grad.gather(dim, index)
+  src: scatter_src_backward(grad.clone(), self, dim, index, reduce)
```

Comment:
Note that this formula has an important side effect.
When it is used for the inplace `scatter_()` (which is the one that everyone use), then we have to save the original value of `self` for this formula. This means that this formula implies a full clone of `self` every time. It would be best if we can replace this `self` by `result` as it would avoid a clone.

Reply:
That's doesn't look very good, indeed.
However, I'm not sure how to do it without `self`. The reason is that when `reduce='multiply'` and one of the values of `src` is zero, we can't really retrieve what the value of `self` was at that position, since `result` at that position will also be zero. Any suggestions?

Reply:
I guess you may solve this by having two overloads: one that doesn't have the reduce argument and one that has a mandatory reduce argument.
Then you can specify formulas for each independently and save different things.

Reply:
I guess you are right. That would also "solve" the issue with ONNX...